openedx/core/djangoapps/content_staging/api.py

```diff
@@ -144,6 +149,46 @@ def _save_static_assets_to_user_clipboard(
             log.exception(f"Unable to copy static file {f.name} to clipboard for component {usage_key}")
 
 
+def save_xblock_to_user_clipboard(block: XBlock, user_id: int, version_num: int | None = None) -> UserClipboardData:
+    """
+    Copy an XBlock's OLX to the user's clipboard.
+    """
+    staged_content = _save_xblock_to_staged_content(block, user_id, CLIPBOARD_PURPOSE, version_num)
+    usage_key = block.usage_key
+
+    # Create/update the clipboard entry
+    (clipboard, _created) = _UserClipboard.objects.update_or_create(
+        user_id=user_id,
+        defaults={
+            "content": staged_content,
+            "source_usage_key": usage_key,
+        },
+    )
+
+    return _user_clipboard_model_to_data(clipboard)
+
+
+def save_xblock_to_user_library_sync(
```

Comment:
The name of this function is pretty confusing, especially if you don't know that there is a class called "UserLibrarySync". There's also not much about this that's really related to libraries or syncing. I'm wondering if you might want to rename it to something like:

```python
def stage_xblock_temporarily(block, user_id):
    """
    "Stage" an XBlock by copying it (and its associated children + static assets)
    into the content staging area. This XBlock can then be accessed and manipulated
    using any of the staged content APIs, before being deleted.
    """
```